Documentation/api.md

```diff
@@ -202,6 +205,44 @@ NamespaceSelector is a selector for selecting either all namespaces or a list of
 
 [Back to TOC](#table-of-contents)
 
+## PodMonitor
+
+PodMonitor defines monitoring for a set of pods.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| metadata | Standard object’s metadata. More info: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata | [metav1.ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#objectmeta-v1-meta) | false |
+| spec | Specification of desired Pod selection for target discrovery by Prometheus. | [PodMonitorSpec](#podmonitorspec) | true |
+
+[Back to TOC](#table-of-contents)
+
+## PodMonitorList
+
+PodMonitorList is a list of PodMonitors.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| metadata | Standard list metadata More info: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata | [metav1.ListMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#listmeta-v1-meta) | false |
+| items | List of PodMonitors | []*[PodMonitor](#podmonitor) | true |
+
+[Back to TOC](#table-of-contents)
+
+## PodMonitorSpec
+
+PodMonitorSpec contains specification parameters for a PodMonitor.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| jobLabel | The label to use to retrieve the job name from. | string | false |
+| targetLabels | TargetLabels transfers labels on the Kubernetes Service onto the target. | []string | false |
+| podTargetLabels | PodTargetLabels transfers labels on the Kubernetes Pod onto the target. | []string | false |
+| endpoints | A list of endpoints allowed as part of this PodMonitor. | [][Endpoint](#endpoint) | true |
```

Comment:
This is a little difficult, at this point I'm not comfortable introducing everything the `Endpoint` struct offers in terms of configuration, for the same reasons why https://github.com/coreos/prometheus-operator/pull/2524 is happening. I'd prefer to have a duplicate struct for this, that does not have the problematic fields that are the goal to be deprecated as part of the overall effort of https://github.com/coreos/prometheus-operator/pull/2524. I'd suggest that we just copy the current `Endpoint` struct and for now remove the `TLSConfig` and basic auth options.

Reply:
Yes agreed. We may also want to be cautious here in future naming as `endpoints` is an existing term in the K8s world and is closely related to the concept of services.

Reply:
Maybe `PodMetricsEndpoint` ?

Reply:
`PodMetricsEndpoint` sounds good